components/hab/src/main.rs

```diff
@@ -1797,6 +1808,123 @@ fn update_svc_load_from_input(m: &ArgMatches<'_>, msg: &mut protocol::ctl::SvcLo
+            let env_var_val = "/val/from/env/var";
+            env_var.set(env_var_val);
+
+            assert_ne!(env_var_val, habitat_common::cli::DEFAULT_BINLINK_DIR);
+            assert_eq!(dest_dir_from_pkg_install(&["origin/pkg", "--binlink"]),
+                       Some(env_var_val.into()),
+                       "with a no-value --binlink arg, the env var value should override the \
+                        default");
+        }
+
+        #[test]
+        fn arg_val_overrides_default() {
+            let env_var = lock_binlink_env_var();
+            env_var.unset();
+
+            let arg_val = "/val/from/args";
+            assert_ne!(arg_val, habitat_common::cli::DEFAULT_BINLINK_DIR);
+            assert_eq!(dest_dir_from_pkg_install(&["origin/pkg", "--binlink", arg_val]),
+                       Some(arg_val.into()),
+                       "The --binlink value should override the default");
+        }
+
+        #[test]
+        fn arg_val_overrides_env_var() {
+            let env_var = lock_binlink_env_var();
+            let env_var_val = "/val/from/env/var";
+            env_var.set(env_var_val);
+            assert_ne!(env_var_val, habitat_common::cli::DEFAULT_BINLINK_DIR);
+
+            let arg_val = "/val/from/args";
+            assert_ne!(arg_val, habitat_common::cli::DEFAULT_BINLINK_DIR);
+
+            assert_eq!(dest_dir_from_pkg_install(&["origin/pkg", "--binlink", arg_val]),
+                       Some(arg_val.into()),
+                       "The --binlink value should override the env var value");
+        }
+
+        #[test]
+        #[should_panic(expected = "required arguments were not provided")]
+        fn binlink_before_pkg_ident_errors() {
+            let env_var = lock_binlink_env_var();
+            env_var.unset();
+
+            dest_dir_from_pkg_install(&["--binlink", "origin/pkg"]);
+        }
+
+        #[test]
+        #[should_panic(expected = "required arguments were not provided")]
+        fn binlink_before_pkg_ident_with_env_var_errors() {
+            let env_var = lock_binlink_env_var();
+            let env_var_val = "/val/from/env/var";
+            env_var.set(env_var_val);
+            assert_ne!(env_var_val, habitat_common::cli::DEFAULT_BINLINK_DIR);
+
+            dest_dir_from_pkg_install(&["--binlink", "origin/pkg"]);
+        }
+
+        fn matches_for_pkg_install<'a>(pkg_install_args: &'a [&'a str]) -> ArgMatches<'a> {
+            let pre_pkg_install_args = &["hab", "pkg", "install"];
+            let app_matches = cli::get(FeatureFlag::empty())
+                .get_matches_from_safe(pre_pkg_install_args.iter().chain(pkg_install_args.iter()))
+                .unwrap(); // Force panics on CLAP errors, so we can use #[should_panic]
+            match app_matches.subcommand() {
+                ("pkg", Some(matches)) => {
+                    match matches.subcommand() {
+                        ("install", Some(m)) => {
+                            println!("{:#?}", m);
```

Comment:
this looks like debug cruft that got unintentionally left?

Reply:
No, that's intentional. All test output is hidden unless the `--nocapture` flag is passed to the test binary. In the case things aren't working as expected, this output is very helpful to see what's going on.